BUG: stats: New XSLOW test failure in test_sampling.py

### Describe your issue.

There is a new XSLOW test failure:

```
$ SCIPY_XSLOW=1 python3 dev.py test -t scipy.stats.tests.test_sampling -m full -j 12
💻  ninja -C /home/warren/repos/git/forks/scipy/build -j12
ninja: Entering directory `/home/warren/repos/git/forks/scipy/build'
[2/338] Generating subprojects/highs/src/HConfig.h with a custom command
Build OK
💻  meson install -C build --only-changed --tags 
runtime,python-runtime,tests,devel
Installing, see meson-install.log...
Installation OK
SciPy from development installed path at: /home/warren/repos/git/forks/scipy/build-install/lib/python3.12/site-packages
Running tests for scipy version:1.15.0.dev0+git20241116.d821271, installed at:/home/warren/repos/git/forks/scipy/build-install/lib/python3.12/site-packages/scipy
============================== test session starts ===============================
platform linux -- Python 3.12.4, pytest-8.2.2, pluggy-1.5.0
rootdir: /home/warren/repos/git/forks/scipy
configfile: pytest.ini
plugins: hypothesis-6.105.1, timeout-2.3.1, xdist-3.6.1, cov-5.0.0, anyio-4.4.0
12 workers [618 items]    
.......................................................................... [ 11%]
.......................................................s...........s.s..... [ 23%]
...............................s...ss..s......s....ss.....s..s..........s. [ 35%]
.............ss.............s....s.......s..s....s.s.........s...........s [ 48%]
............ssss.ss.........ss.....s....s....sss.s........................ [ 60%]
..........x...................F...............x...x.....x......x......ss.. [ 72%]
.xx.x..x...s.................x..........xxx...s......x.....x.............. [ 83%]
................s...s.s...ssssssss..ss...s................................ [ 95%]
.........................                                                  [100%]
==================================== FAILURES ====================================
_ TestNumericalInverseHermite.test_basic_all_scipy_dists[dpareto_lognorm-shapes15] _
[gw1] linux -- Python 3.12.4 /home/warren/py3.12.4/bin/python3
scipy/stats/tests/test_sampling.py:1110: in test_basic_all_scipy_dists
    self.basic_test_all_scipy_dists(distname, shapes)
        distname   = 'dpareto_lognorm'
        self       = <scipy.stats.tests.test_sampling.TestNumericalInverseHermite object at 0x72e14e5abda0>
        shapes     = (3, 1.2, 1.5, 2)
scipy/stats/tests/test_sampling.py:1095: in basic_test_all_scipy_dists
    fni = NumericalInverseHermite(dist)
        dist       = <scipy.stats._distn_infrastructure.rv_continuous_frozen object at 0x72e145bb74d0>
        distname   = 'dpareto_lognorm'
        fail_dists = {'beta', 'gausshyper', 'genhyperbolic', 'geninvgauss', 'invgauss', 'kappa4', ...}
        self       = <scipy.stats.tests.test_sampling.TestNumericalInverseHermite object at 0x72e14e5abda0>
        shapes     = (3, 1.2, 1.5, 2)
        slow_dists = {'ksone', 'kstwo', 'levy_stable', 'skewnorm'}
unuran_wrapper.pyx:1993: in scipy.stats._unuran.unuran_wrapper.NumericalInverseHermite.__cinit__
    ???
unuran_wrapper.pyx:443: in scipy.stats._unuran.unuran_wrapper.Method._set_rng
    ???
E   scipy.stats._unuran.unuran_wrapper.UNURANError: [objid: HINV.025] 50 : CDF not increasing => (generator) (possible) invalid data
============================ short test summary info =============================
FAILED scipy/stats/tests/test_sampling.py::TestNumericalInverseHermite::test_basic_all_scipy_dists[dpareto_lognorm-shapes15] - scipy.stats._unuran.unuran_wrapper.UNURANError: [objid: HINV.025] 50 : CDF no...
============= 1 failed, 547 passed, 55 skipped, 15 xfailed in 4.57s ==============
```

### Reproducing Code Example

```python
N/A
```


### Error message

```shell
See above.
```


### SciPy/NumPy/Python version and system information

```shell
1.15.0.dev0+git20241116.d821271 2.1.3 sys.version_info(major=3, minor=12, micro=4, releaselevel='final', serial=0)
Build Dependencies:
  blas:
    detection method: pkgconfig
    found: true
    include directory: /usr/include/x86_64-linux-gnu/openblas-pthread/
    lib directory: /usr/lib/x86_64-linux-gnu/openblas-pthread/
    name: openblas
    openblas configuration: USE_64BITINT= DYNAMIC_ARCH=1 DYNAMIC_OLDER=1 NO_CBLAS=
      NO_LAPACK= NO_LAPACKE=1 NO_AFFINITY=1 USE_OPENMP=0 generic MAX_THREADS=64
    pc file directory: /usr/lib/x86_64-linux-gnu/pkgconfig
    version: 0.3.20
  lapack:
    detection method: pkgconfig
    found: true
    include directory: /usr/include/x86_64-linux-gnu/openblas-pthread/
    lib directory: /usr/lib/x86_64-linux-gnu/openblas-pthread/
    name: openblas
    openblas configuration: USE_64BITINT= DYNAMIC_ARCH=1 DYNAMIC_OLDER=1 NO_CBLAS=
      NO_LAPACK= NO_LAPACKE=1 NO_AFFINITY=1 USE_OPENMP=0 generic MAX_THREADS=64
    pc file directory: /usr/lib/x86_64-linux-gnu/pkgconfig
    version: 0.3.20
  pybind11:
    detection method: config-tool
    include directory: unknown
    name: pybind11
    version: 2.13.2
Compilers:
  c:
    commands: cc
    linker: ld.bfd
    name: gcc
    version: 11.4.0
  c++:
    commands: c++
    linker: ld.bfd
    name: gcc
    version: 11.4.0
  cython:
    commands: cython
    linker: cython
    name: cython
    version: 3.0.10
  fortran:
    commands: gfortran
    linker: ld.bfd
    name: gcc
    version: 11.4.0
  pythran:
    include directory: ../../../../../py3.12.4/lib/python3.12/site-packages/pythran
    version: 0.16.1
Machine Information:
  build:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
  cross-compiled: false
  host:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
Python Information:
  path: /home/warren/py3.12.4/bin/python3
  version: '3.12'
```

--- a/scipy/stats/_continuous_distns.py
+++ b/scipy/stats/_continuous_distns.py
@@ -1846,7 +1846,7 @@ class dpareto_lognorm_gen(rv_continuous):
         \left( R(y_1) + R(y_2) \right)
 
     where :math:`R(t) = \frac{1 - \Phi(t)}{\phi(t)}`,
-    :math:`phi` and :math:`Phi` are the normal PDF and CDF, respectively,
+    :math:`\phi` and :math:`\Phi` are the normal PDF and CDF, respectively,
     :math:`y_1 = \alpha \sigma - \frac{\log x - \mu}{\sigma}`,
     and :math:`y_2 = \beta \sigma + \frac{\log x - \mu}{\sigma}`
     for real numbers :math:`x` and :math:`\mu`, :math:`\sigma > 0`,

--- a/scipy/stats/_continuous_distns.py
+++ b/scipy/stats/_continuous_distns.py
@@ -1923,20 +1923,24 @@ class dpareto_lognorm_gen(rv_continuous):
         return out[()]
 
     def _logcdf(self, x, u, s, a, b):
-        log_y, m = np.log(x), u  # compare against [1] Eq. 2
-        z = (log_y - m) / s
-        x1 = a * s - z
-        x2 = b * s + z
-        t1 = self._logPhi(z)
-        t2 = self._logphi(z)
-        t3 = (np.log(b) + self._logR(x1))
-        t4 = (np.log(a) + self._logR(x2))
-        t1, t2, t3, t4, one = np.broadcast_arrays(t1, t2, t3, t4, 1)
-        # t3 can be smaller than t4, so we have to consider log of negative number
-        # This would be much simpler, but `return_sign` is available, so use it?
-        # t5 =  sc.logsumexp([t3, t4 + np.pi*1j])
-        t5, sign =  sc.logsumexp([t3, t4], b=[one, -one], axis=0, return_sign=True)
-        return sc.logsumexp([t1, t2 + t5 - np.log(a + b)], b=[one, -one*sign], axis=0)
+        with np.errstate(invalid='ignore', divide='ignore'):
+            log_y, m = np.log(x), u  # compare against [1] Eq. 2
+            z = (log_y - m) / s
+            x1 = a * s - z
+            x2 = b * s + z
+            t1 = self._logPhi(z)
+            t2 = self._logphi(z)
+            t3 = (np.log(b) + self._logR(x1))
+            t4 = (np.log(a) + self._logR(x2))
+            t1, t2, t3, t4, one = np.broadcast_arrays(t1, t2, t3, t4, 1)
+            # t3 can be smaller than t4, so we have to consider log of negative number
+            # This would be much simpler, but `return_sign` is available, so use it?
+            # t5 =  sc.logsumexp([t3, t4 + np.pi*1j])
+            t5, sign =  sc.logsumexp([t3, t4], b=[one, -one], axis=0, return_sign=True)
+            temp = [t1, t2 + t5 - np.log(a + b)]
+            out = np.asarray(sc.logsumexp(temp, b=[one, -one*sign], axis=0))
+        out[x == 0] = -np.inf
+        return out[()]
 
     def _logsf(self, x, u, s, a, b):
         return _log1mexp(self._logcdf(x, u, s, a, b))

